File: guides/micronaut-security-session-database-authentication/java/src/main/java/example/micronaut/entities/Role.java
package example.micronaut.entities;

import io.micronaut.core.annotation.Nullable;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import jakarta.validation.constraints.NotBlank;

@MappedEntity // <1>
public record Role(@Nullable
                   @Id // <2>
                   @GeneratedValue // <3>
                   Long id,
                   @NotBlank // <4>
                   String authority) {
}

File: guides/micronaut-security-session-database-authentication/python/src/example/micronaut/entities/role.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotBlank
from micronaut.data.annotation import GeneratedValue, Id, MappedEntity
from micronaut.serde.annotation import Serdeable


@dataclass
@Serdeable
@MappedEntity  # <1>
class Role:
    authority: Annotated[str, NotBlank]  # <2>
    id: Annotated[int | None, Id, GeneratedValue] = None  # <3> <4>
